docs/SERVICE-MESH.md

```diff
@@ -0,0 +1,242 @@
+# Open Data Hub (ODH) Installation Guide with OpenShift Service Mesh (OSSM)
+
+This guide will walk you through the installation of Open Data Hub with OpenShift Service Mesh.
+
+## Prerequisites
+
+* OpenShift cluster
+* Command Line Interface (CLI) tools
+  * `kubectl`
+  * `operator-sdk` v1.24.1 (until operator changes are merged - to build new bundles)
+
+* Pre-installed operators
+  * Openshift Service Mesh
+  * Authorino
+  * Open Data Hub
+
+* Service Mesh Control Plane configured
+  
+### Check Installed Operators
+
+You can use the following command to verify that all required operators are installed:
+
+```sh
+kubectl get operators | awk -v RS= '/servicemesh/ && /opendatahub/ && /authorino/ {exit 0} {exit 1}' || echo "Please install all required operators."
+```
+
+#### Install Required Operators
+
+The `createSubscription` function can be used to simplify the installation of required operators:
+
+```sh
+createSubscription() {
+  local name=$1
+  local source=${2:-"redhat-operators"}
+  local channel=${3:-"stable"}
+
+  echo  "Create Subscription resource for $name"
+  eval "kubectl apply -f - <<EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: $name
+  namespace: openshift-operators
+spec:
+  channel: $channel
+  installPlanApproval: Automatic
+  name: $name
+  source: $source
+  sourceNamespace: openshift-marketplace
+EOF"    
+}
+```
+
+You can use the function above to install all required operators:
+
+```sh
+createSubscription "servicemeshoperator"
+createSubscription "authorino-operator" "community-operators" "alpha"
+```
+
+or use Openshift Console to click through the UI.
+
+To run pre-built image of this PR simply execute:
+```sh
+operator-sdk run bundle quay.io/maistra-dev/opendatahub-operator-bundle:v1.0.0-service-mesh --namespace openshift-operators --timeout 5m0s
+```
+
+or go to [README.md#Deploying operator locally](../README.md#deployment) to learn how to build and test changes done locally.
+
+> **Warning**
+>
+> You may need to manually update the installation of the Authorino operator via the Installed Operators tab in the OpenShift Console.
+
+
+> **Warning**
+>
+> Please ensure that the Service Mesh Control Plane is properly configured as we apply patches to it. It is assumed that the installation has already been done.
+
+
+For example, the following commands configure a slimmed-down profile:
+
+```sh
+kubectl create ns istio-system
+kubectl apply -n istio-system -f -<<EOF
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: minimal
+spec:
+  version: v2.4
+  tracing:
+    type: None
+  addons:
+    prometheus:
+      enabled: false
+    grafana:
+      enabled: false
+    jaeger:
+      name: jaeger
+    kiali:
+      name: kiali
+      enabled: false
+EOF
+```
```

Comment:
I sanity tested the spec, with the result being that the SMCP deployed fine with the above spec.

Note: "name: kiali" is not needed, the SMCP deployed fine without the name.